### PR TITLE
Fix/403 readony users mbp 529

### PR DESCRIPTION
**Pull request by radireddy**
Fixed below issues

1. manager shows error page with 403 error when read only user tries to access kubernetes dsshboard page
2. add missing k8s status like suspending, suspended, user_error, user_quota_error and reopening

MBP-411, MBP-410, MBP-529

### CODE REVIEW

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/service/translations/Messages_fr_FR.json [hunk 1]
```diff
@@ -17,6 +17,11 @@
   "kube_service_cluster_status_UPDATING": "Mise à jour",
   "kube_service_cluster_status_DELETING": "En cours de suppression",
   "kube_service_cluster_status_RESETTING": "En cours de réinitialisation",
+  "kube_service_cluster_status_SUSPENDING": "Suspension en cours",
+  "kube_service_cluster_status_REOPENING": "Ouverture en cours",
```
Comment (FredericEspiau) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/service/translations/Messages_fr_FR.json]:
```suggestion
  "kube_service_cluster_status_REOPENING": "Réouverture en cours",
```

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/service/translations/Messages_fr_FR.json [hunk 2]
```diff
@@ -17,6 +17,11 @@
...
+  "kube_service_cluster_status_SUSPENDED": "Suspendu",
+  "kube_service_cluster_status_USER_ERROR": "Erreur d'utilisateur",
```
Comment (FredericEspiau) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/service/translations/Messages_fr_FR.json]:
```suggestion
  "kube_service_cluster_status_USER_ERROR": "En erreur",
```

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/service/translations/Messages_fr_FR.json [hunk 3]
```diff
@@ -17,6 +17,11 @@
...
+  "kube_service_cluster_status_USER_QUOTA_ERROR": "Erreur de quota utilisateur",
```
Comment (FredericEspiau) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/service/translations/Messages_fr_FR.json]:
```suggestion
  "kube_service_cluster_status_USER_QUOTA_ERROR": "Problème de quota",
```

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/service/service.controller.js [hunk 4]
```diff
@@ -37,7 +40,14 @@ export default class KubernetesServiceCtrl {
...
-    // Set yml extension manually as there is no MIME type yet
-    this.CucControllerHelper.constructor.downloadContent({ fileContent: this.kubernetesConfig.content, fileName: `${this.kubernetesConfig.fileName}.yml` });
+    this.loadingKubeConfig = true;
+    return this.getKubeConfig()
+      .then((config) => {
+        // Set yml extension manually as there is no MIME type yet
+        this.CucControllerHelper.constructor
+          .downloadContent({ fileContent: config.content, fileName: `${config.fileName}.yml` });
+      })
+      .catch(error => this.CucServiceHelper.errorHandler('kube_service_file_error')(error))
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/service/service.controller.js]: We should probbaly uniformize error handling by using `CucCloudMessage` everywhere. That way if we want to change we only have to worry about one migration :)

**Comment by ovh-cds** [comment]
CDS Report telecom-task#4315.0 ✘  
* 
  * Build CA ✔ 
  * Build EU ✘  
  * Build US ✔

File: packages/manager/apps/public-cloud/src/index.js [hunk 5]
```diff
@@ -77,7 +77,9 @@ angular
   })
   .run(/* @ngInject */ ($state) => {
     $state.defaultErrorHandler((error) => {
-      if (error.type === RejectType.ERROR) {
+      console.log('error', error);
```
Comment (FredericEspiau) [inline packages/manager/apps/public-cloud/src/index.js]: ?

Comment (radireddy) [reply packages/manager/apps/public-cloud/src/index.js]: just added to check something in staging, its removed now

Comment (radireddy) [reply packages/manager/apps/public-cloud/src/index.js]: as of now readonly user can only be tested in staging

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/service/translations/Messages_fr_FR.json [hunk 6]
```diff
@@ -6,7 +6,7 @@
   "kube_service_description_reset": "Vous pouvez réinitialiser la configuration de votre cluster Kubernetes à tout moment.",
   "kube_service_offer_beta": "Service managé Kubernetes (gratuit)",
   "kube_service_file": "Fichier kubeconfig",
-  "kube_service_file_error": "Une erreur est survenue lors du chargement du fichier de configuration",
+  "kube_service_file_error": "Une erreur est survenue lors du chargement du fichier de configuration: {{ message }}",
```
Comment (antleblanc) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/service/translations/Messages_fr_FR.json]:
```suggestion
  "kube_service_file_error": "Une erreur est survenue lors du chargement du fichier de configuration : {{ message }}",
```

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/service/service.controller.js [hunk 7]
```diff
@@ -37,7 +38,14 @@ export default class KubernetesServiceCtrl {
...
-    // Set yml extension manually as there is no MIME type yet
-    this.CucControllerHelper.constructor.downloadContent({ fileContent: this.kubernetesConfig.content, fileName: `${this.kubernetesConfig.fileName}.yml` });
+    this.loadingKubeConfig = true;
+    return this.getKubeConfig()
+      .then((config) => {
+        // Set yml extension manually as there is no MIME type yet
+        this.CucControllerHelper.constructor
+          .downloadContent({ fileContent: config.content, fileName: `${config.fileName}.yml` });
+      })
+      .catch(error => this.CucCloudMessage.error([this.$translate.instant('kube_service_file_error'), (error.data && error.data.message) || ''].join(' ')))
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/service/service.controller.js]:
```suggestion
      .catch(error => this.CucCloudMessage.error(`${this.$translate.instant('kube_service_file_error')} ${_.get(error, 'data.message})
```

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/service/service.html [hunk 8]
```diff
@@ -21,7 +21,7 @@
                             data-ng-class="{
                                     'oui-status_success' : $ctrl.cluster.status === $ctrl.STATUS.READY,
                                     'oui-status_warning' : $ctrl.Kubernetes.constructor.isProcessing($ctrl.cluster.status),
-                                    'oui-status_error' : $ctrl.cluster.status === $ctrl.STATUS.ERROR
+                                    'oui-status_error' : $ctrl.Kubernetes.constructor.isError($ctrl.cluster.status) || $ctrl.cluster.status === $ctrl.STATUS.SUSPENDED,
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/service/service.html]: Why not use the same method as for `ERROR`status with a `isSuspended($ctrl.cluster.status)`

File: packages/manager/modules/pci/src/projects/project/kubernetes/details/service/service.html [hunk 9]
```diff
@@ -88,8 +88,13 @@
                             </oui-popover-content>
                         </oui-popover>
                     </dt>
-                    <dd class="oui-tile__description">
-                        <oui-button data-variant="link" data-ng-click="$ctrl.downloadConfigFile()" data-text="{{:: $ctrl.kubernetesConfig.fileName }}"></oui-button>
+                    <dd class="oui-tile__description d-flex align-items-center">
+                        <oui-button data-variant="link"
+                            data-ng-click="$ctrl.downloadConfigFile()"
+                            data-disabled="$ctrl.loadingKubeConfig"
+                            data-text="{{:: $ctrl.CONFIG_FILENAME }}">
```
Comment (marie-j) [inline packages/manager/modules/pci/src/projects/project/kubernetes/details/service/service.html]: I think `data-text` is deprecated

**Comment by antleblanc** [comment]
@radireddy Could you please resolve existing conflicts in order to be able to merge this Pull request?

Thanks!

**Comment by radireddy** [comment]
@antleblanc all conflicts are resolved and all review comments are taken care.

**Pull request merged**